src/open_inwoner/openzaak/cases.py

```diff
@@ -303,8 +303,9 @@ def fetch_single_result(result_url: str) -> Optional[Resultaat]:
     return result
 
 
-def connect_case_with_document(case_url: str, document_url: str) -> dict:
+def connect_case_with_document(case_url: str, document_url: str) -> Optional[dict]:
```

Comment:
If there is a normal return value but also a bare `return` (eg: None) then the return type should be `Optional`